package/MDAnalysis/auxiliary/EDR.py

```diff
@@ -0,0 +1,358 @@
+GROMACS during simulations and contain the time-series energy data of the
+system.
+
+pyedr_ is a Python package that reads EDR binary files and returns them
+human-readable form as a dictionary of NumPy arrays. It is used by the EDR
+auxiliary reader to parse EDR files. As such, a dictionary with string keys and
+numpy array values is loaded into the :class:`EDRReader`. It is basically a
+Python-based version of the C++ code in GROMACS_.
+
+The EDR auxiliary reader takes the output from pyedr and loads the energy data
+as auxiliary data into :class:`MDAnalysis.core.universe.Universe`. Standalone
+usage is also possible, where the energy terms are extracted without
+associating them with the trajectory, for example, to allow easy plotting of
+the energy terms.
+
+For adding EDR data, 2-3 arguments need to be provided to
+:func:`MDAnalysis.coordinates.base.add_auxiliary`: `auxname`, `auxdata`, and
+(optionally) `auxterm`. `auxterm`s, if provided need to exactly match the names
+of EDR terms to be added. When an auxterm is provided, `auxname` can be freely
+chosen. If an `auxterm` is not provided, `auxname` must instead exactly match
+EDR terms. `auxdata` can be either the path to an EDR file, or an instance of
+:class:`MDAnalysis.auxiliary.EDR.EDRReader`.
+
+Examples::
+
+    import MDAnalysis as mda
+    from MDAnalysisTests.datafiles import AUX_EDR, AUX_EDR_TPR, AUX_EDR_XTC
+    import matplotlib.pyplot as plt
+
+The EDRReader can be used to add auxiliary data from EDR files to
+trajectories::
+
+    # Create a Universe
+    In [1]: u = mda.Universe(AUX_EDR_TPR, AUX_EDR_XTC)
+    # Open the EDR file with an EDRReader
+    In [2]: aux = mda.auxiliary.EDR.EDRReader(AUX_EDR)
+    # See which terms are present in the file
+    In [3]: aux.terms
+    Out[3]: ['Time', 'Bond', 'Angle', ...]
+    # add the Potential data under the name epot
+    In [4]: u.trajectory.add_auxiliary("epot", aux, "Potential")
+    # add the Angle energy without stating an `auxterm`
+    In [5]: u.trajectory.add_auxiliary("Angle", aux)
+    # access the value of Potential and Angle for a timestep
+    In [6]: u.trajectory.ts.aux.epot
+    Out[6]: -525164.0625
+    In [7]: u.trajectory.ts.aux.Angle
+    Out[7]: 3764.52734375
+    # dictionary style accession also works
+    In [8]: u.trajectory.ts.aux["epot"]
+    Out[8]: -525164.0625
+
+
+Note: Some `auxterms` have spaces in their names. Unless an `auxname` without a
+space is provided, these terms will not be accessible via the attribute syntax.
+Only the dictionary syntax will work in that case.
+
+
+Adding multiple or all terms at the same time is possible as such::
+
+    # add list of values
+    u.trajectory.add_auxiliary(["Bond", "Angle"], aux)
+    # add list of values, provide an `auxterm`
+    u.trajectory.add_auxiliary(["epot", "ekin"], aux, ["Potential",
+                                                       "Kinetic En."]
+    # add all values
+    u.trajectory.add_auxiliary("*", aux)
+
+
+
```

Comment:
pedantic: are the three empty lines necessary here?

Reply:
They are not, changed now